pkg/controller/atlasfederatedauth/atlasfederated_auth.go

```diff
@@ -110,8 +117,23 @@ func (r *AtlasFederatedAuthReconciler) ensureIDPSettings(ctx context.Context, fe
 	return workflow.OK()
 }
 
-func federatedSettingsAreEqual(operator, atlas *mongodbatlas.FederatedSettingsConnectedOrganization) bool {
+func federatedSettingsAreEqual(operator, atlas *admin.ConnectedOrgConfig) bool {
 	operator.UserConflicts = nil
 	atlas.UserConflicts = nil
 	return cmp.Diff(operator, atlas) == ""
 }
+
+func GetIdentityProviderForFederatedSettings(ctx context.Context, atlasClient *admin.APIClient, fedSettings *admin.OrgFederationSettings) (*admin.FederationIdentityProvider, error) {
+	identityProviders, _, err := atlasClient.FederatedAuthenticationApi.ListIdentityProviders(ctx, fedSettings.GetId()).Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, identityProvider := range identityProviders.GetResults() {
+		if identityProvider.GetOktaIdpId() == fedSettings.GetIdentityProviderId() {
+			return &identityProvider, nil
```

Comment:
what if there are more matches?

Reply:
We can double check, but if I'm not wrong, as we don't control the whole workflow of FedAuth, the stage we can do something, an idp must be configured already for org/fedSettings

Reply:
Yes, the IDP has to be configured before using FederatedAuth settings in the AKO.